### PR TITLE
refactor(react): convert `Search` to TypeScript

### LINKED ISSUE #12553: Add TypeScript types to Search

### PR DESCRIPTION
**Pull request by metonym**
Closes #12553

#### Changelog

**Changed**

- refactor(react): convert `Search` to TypeScript

#### Testing / Reviewing

{{ Add descriptions, steps or a checklist for how reviewers can verify this PR works or not }}

### CODE REVIEW

**Comment by francinelucca** [comment]
Hey @metonym 👋🏼, thanks for contributing this! One minor ask - instead of renaming the .js file to .tsx, could you actually rename it using git?

`git mv packages/react/src/components/Search/Search.js packages/react/src/components/Search/Search.tsx`

This way the diff will clearly show the file extension change but also the proper unified diff showing your changes within file.

Right now it's hard to tell what you've added to the file vs was was in it to begin with.

**Review by tay1orjones — CHANGES_REQUESTED** [review]
Just popping a "requested changes" on here in reference to Francine's comment

**Comment by metonym** [comment]
@francinelucca Thanks for the feedback. When I ran the `git mv` command, it successfully displayed it as [one change](https://github.com/carbon-design-system/carbon/pull/12998/commits/de826447e1e62b97e7754f63835a5628863a3d34). However, after applying the actual changes to `Search.tsx`, it appears as two separate files again.

File: packages/react/src/components/Search/Search.tsx [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+
+  /**
+   * Specify an optional className to be applied to the container node
+   */
+  className?: string;
+
+  /**
+   * Specify a label to be read by screen readers on the "close" button
+   */
+  closeButtonLabelText?: string;
```
Comment (francinelucca) [inline packages/react/src/components/Search/Search.tsx]:
```suggestion
  /**
   * Specify a label to be read by screen readers on the "close" button
   * @default "Clear search input"
   */
  closeButtonLabelText?: string;
```

**Review by francinelucca — CHANGES_REQUESTED** [review]
Minor suggestion, other than that LGTM

**Comment by aledavila** [comment]
Bump. Looks like there are some merge conflicts @metonym to fix ! TIA

**Review by tay1orjones — APPROVED** [review]

**Comment by metonym** [comment]
Thanks for the nudge – resolved conflicts in this commit: adf2045cb4c3ad6ed35d6b72b9ac437ebd4c9bd4

**Comment by francinelucca** [comment]
Looks like a couple tests are failing for search now @metonym

**Comment by aledavila** [comment]
Hey @metonym. Thanks so much for working on this contribution! We’d really like to get this merged but with the current test failures it’s not able to merge. If we can resolve those in the next few days we can merge, otherwise we’ll close this for inactivity.

**Comment by metonym** [comment]
Closed by #13533

**Pull request closed without merging**